File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */
    kthSmallest(root, k) {
        if(root == null) return null
        let result = []
        function inorder(node){
            if(!node) return
            inorder(node.left)
            result.push(node.val)
            inorder(node.right)
        }
        inorder(root)
        return result[k-1]
    }
}

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
     List<Integer> l1;
     public void inOrder(TreeNode root) {
        if(root == null) return;
        if(root.left != null) inOrder(root.left);
        l1.add(root.val);
        if(root.right != null) inOrder(root.right);
     }
    public List<Integer> inorderTraversal(TreeNode root) {
        l1 = new ArrayList<>();
        inOrder(root);
        return l1;
    }
}